pkg/instance/repository.go

```diff
@@ -75,8 +75,61 @@ func (r repository) FindDeploymentById(id uint) (*model.Deployment, error) {
 	return deployment, nil
 }
 
+func (r repository) FindDeploymentInstanceById(id uint) (*model.DeploymentInstance, error) {
+	var instance *model.DeploymentInstance
+	err := r.db.
+		Preload("GormParameters").
+		First(&instance, id).Error
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, errdef.NewNotFound("instance not found by id: %d", id)
```

Comment:
Should we also use the standard `"failed to ..."` error message prefix here?

Reply:
Yeah, there's a mix of `X not found ...` and `failed to ...` error messages in both new and old code. Only the latter is mentioned in [the style docs here](https://github.com/dhis2-sre/im-doc/tree/master/dev#errors), so we should either stick to it (and even go back and apply it across all error messages) or update the doc. 👀